### DISCUSSION TITLE
Use codeowners to suggest reviewers, NOT automatically assign

### DISCUSSION THREAD

**Discussion by smonette** [community/community#22866] [New to GitHub]
<p>Hi there!</p>
<p>My team has recently added codeowners. In the past few weeks we’ve found that automatically assigning reviewers when someone creates a PR is disruptive to a lot of individual workflows.</p>
<p>How can we use codeowners to populate the “Suggestion” dropdown, and allow teammates to manually select when they’re ready for review?</p>
<p>Thanks!</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>There isn’t an option to make CODEOWNERS work in the way you describe. The Suggestion dropdown is populated based on who has modified certain files in the past. If you want a system that works similarly to the way you describe, you would have to create some other kind of owners system and the tooling to request reviews from the owners. You could perhaps use <a href="https://probot.github.io" rel="nofollow noopener">Probot</a> for something like that. You may want to look at the <a href="https://probot.github.io/apps/auto-assign/" rel="nofollow noopener">Auto Assign Probot app</a> for inspiration.</p>

**Comment by smonette** [comment]
<p>Thank you so much!</p>
<p>The Auto Assign Probot looks perfect for our needs! </p>

**Comment by jmasukawa** [comment]
My team was having this same issue, and I found this thread from a Google search. Figure I'd update with our solution: have authors use the (now available) "draft PR" feature. See: https://github.blog/2019-02-14-introducing-draft-pull-requests/

This bit in particular: "Also, if you have a CODEOWNERS file in your repository, a draft pull request will suppress notifications to those reviewers until it is marked as ready for review."

**Reply by oerp-odoo** [reply]
@jmasukawa but isn't that just gonna delay the problem? In our case, we have team of developers that is owner of the code. So it automatically assigns one of the developers as reviewer. But sometimes we want to select specific developer instead of being selected by code owner feature. So you notify someone randomly and then select person you want. Kind of clunky way.

Would be nice to have  some option, to check "ignore code owners for this PR" or something like that.

**Reply by jmasukawa** [reply]
> But sometimes we want to select specific developer instead of being selected by code owner feature.

@oerp-odoo Oh, for that scenario then my suggestion will not help.

My problem was not that CODEOWNERS could pick an "incorrect" reviewer sometimes 

The problem I had was that CODEOWNERS assignment was happening too early (automatically on PR creation).

We run PR-time cloud build checks, linting checks, run tests, etc. So I wanted to create a PR without assignment, so I can make sure those checks passed. Then add reviewers after. So delaying the process was fine in my case.
